File: lang/c/test/vac.fun/algorithms/sorting/SelectionSortTest.c
#include <vac.fun/algorithms/sorting/SelectionSort.h>
#include <vac.fun/util/SequenceBuilder.h>
#include <vac.fun/util/Sequences.h>
#include <vac.fun/util/TestRunner.h>

bool testSelectionSort(void) {
    size_t size = 32768;

    int arr[size];
    SequenceBuilder_packRandom(arr, size);

    int checksum = Sequences_parityChecksum(arr, size);

    SelectionSort_sort(arr, size);

    if (Sequences_parityChecksum(arr, size) != checksum) {
        return false;
    }

    return Sequences_isSorted(arr, size);
}

int main(void) {
    TestRunner_parseTest(testSelectionSort());
}

File: lang/go/test/vac.fun/algorithms/sorting/SelectionSort/SelectionSortTest.go
package main

import (
    "vac.fun/algorithms/sorting/SelectionSort"
    "vac.fun/util/SequenceBuilder"
    "vac.fun/util/Sequences"
    "vac.fun/util/TestRunner"
)

func testSelectionSort() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackRandom(arr)

    checksum := Sequences.ParityChecksum(arr)

    SelectionSort.Sort(arr)

    if Sequences.ParityChecksum(arr) != checksum {
        return false
    }

    return Sequences.IsSorted(arr)
}

func main() {
    TestRunner.ParseTest(testSelectionSort())
}
